src/aura/BGE_DataImportBatchEntry/BGE_DataImportBatchEntryController.js

```diff
@@ -28,29 +28,49 @@
     handleMessage: function (component, event, helper) {
         let message = event.getParam('message');
         let channel = event.getParam('channel');
-        let info;
+        let relevantChannels = ['onSuccess', 'onCancel', 'setDonorType', 'hideFormSpinner', 'showFormSpinner', 'onError'];
 
-        if (message) {
-            if (message.batchId !== component.get('v.recordId')) {
-                return;
-            }
-            info = message.info || message;
+        // Check if channel is in relevant channels or if there is no message and return
```

Comment:
Is the addition of the channels enough to avoid having to use the batchId? Could there be a another tab open that sends in a message for one of these relevant channels but is intended for a different batch?